src/main/java/stobberi/components/Storage.java

```diff
@@ -0,0 +1,119 @@
+package stobberi.components;
+
+import stobberi.stobberiException.NotPossibleDurationStobberiException;
+import stobberi.task.Deadline;
+import stobberi.task.Event;
+import stobberi.task.Task;
+import stobberi.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * The {@code Storage} class handles reading and writing {@code Task} objects to and from a file.
+ * It supports reading tasks of different types (Todo, Deadline, Event) from a file and saving tasks
+ * back to the file. This class is responsible for the persistence of task data.
+ */
+public class Storage {
```

Comment:
Good job on using clear English names for all your class and method names, the names are also clear and not misleading!

Reply:
Thank you!